src/HealthChecks.Uris/UriHealthCheck.cs

```diff
@@ -44,7 +44,19 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
                         requestMessage.Headers.Add(header.Name, header.Value);
                     }
 
-                    var response = await httpClient.SendAsync(requestMessage);
+                    HttpResponseMessage response;
+                    if (_options.Timeout != TimeSpan.Zero)

```

Comment:
is _options.Timeout or item.Timeout? You need to get _options.Timeout as default parameter if is not override with specified UriOptions. Is the same pattern we have with defaultstatuscode httpmethod etc